utils/js/websocketosaurus.html

```diff
@@ -1,285 +1,44 @@
-					lblState.innerHTML = stateStr
-				} else {
-					debug("WebSocket is null");
-					lblState.innerHTML = "no-socket"
-				}
-			}
-
-			function callRpcMethod(shv_path, method, params)
-			{
-				let rq = new RpcMessage();
-				rq.setRequestId(request_id++);
-				if(shv_path)
-					rq.setShvPath(shv_path);
-				rq.setMethod(method);
-				if(params)
-					rq.setParams(RpcValue.fromCpon(params));
-				sendRpcMessage(rq);
-			}
-
-			function sendRpcMessage(rpc_msg)
-			{
-				if(websocket && websocket.readyState == 1) {
-					debug("sending rpc message:", rpc_msg.toString())
-					if(isUseCpon)
-						var msg_data = new Uint8Array(rpc_msg.toCpon());
-					else
-						var msg_data = new Uint8Array(rpc_msg.toChainPack());
-
-					let wr = new ChainPackWriter();
-					wr.writeUIntData(msg_data.length + 1)
-					let dgram = new Uint8Array(wr.ctx.length + 1 + msg_data.length)
-					let ix = 0
-					for (let i = 0; i < wr.ctx.length; i++)
-						dgram[ix++] = wr.ctx.data[i]
-
-					if(isUseCpon)
-						dgram[ix++] = Cpon.ProtocolType
-					else
-						dgram[ix++] = ChainPack.ProtocolType
-
-					for (let i = 0; i < msg_data.length; i++)
-						dgram[ix++] = msg_data[i]
-					debug("sending " + dgram.length + " bytes of data")
-					websocket.send(dgram.buffer)
-				}
-			}
-
-			function datatoRpcValue(buff)
-			{
-				let rd = new ChainPackReader(new UnpackContext(buff));
-				let len = rd.readUIntData()
-				let proto = rd.ctx.getByte();
-				if(proto == Cpon.ProtocolType)
-					rd = new CponReader(rd.ctx)
-				else
-					rd = new ChainPackReader(rd.ctx)
-				let rpc_val = rd.read();
-				//debug("msg buff len " + buff.byteLength + " offset: " + offset)
-				//return String.fromCharCode.apply(null, new Uint8Array(buff, offset));
-				return rpc_val;
-			}
-		</script>
-	</body>
-</html>
+<!DOCTYPE html>
+<html>
+	<head>
+		<title>WebSocketoSaurus</title>
+		<meta charset="utf-8">
+		<script type="text/javascript" src="../../libshvchainpack/js/rpcvalue.js"></script>
+		<script type="text/javascript" src="../../libshvchainpack/js/cpcontext.js"></script>
+		<script type="text/javascript" src="../../libshvchainpack/js/chainpack.js"></script>
+		<script type="text/javascript" src="../../libshvchainpack/js/cpon.js"></script>
+		<script type="text/javascript" src="../../libshvchainpack/js/bint.js"></script>
+		<script type="text/javascript" src="../../libshvchainpack/js/rpcmessage.js"></script>
+		<script type="text/javascript" src="../../libshvchainpack/js/ws_client.js"></script>
+	</head>
+	<body>
+		<h1>WebSocketoSaurus</h1>
+		<h2>WebSocket Shv Broker test Client</h1>
+		<p>
+			Uri <input type="text" id="edUri" style="width:50em;"/>
+		</p>
+		<p>
+			user <input type="text" id="edUser"/>
+			password <input type="password" id="edPassword"/>
+		</p>
+		<p>
+			<button onClick="initWebSocket(true);">Connect ChainPack</button>
+			<button onClick="initWebSocket(false);">Connect Cpon</button>
+			<button onClick="stopWebSocket();">Disconnect</button>
```

Comment:
where is `lblState`?

Reply:
I thought it wasn't very useful, so I removed it. I could readd it, but it's kind of a hassle now, because the underlying `WebSocket` and its `on` methods are handled by the client library.